test/unittests/floating_point_utils_test.cpp

```diff
@@ -7,6 +7,42 @@
 
 using namespace fizzy::test;
 
+TEST(floating_point_utils, fp_default)
+{
+    EXPECT_EQ(FP32{}.as_uint(), 0);
+    EXPECT_EQ(FP32{}.as_float(), 0.0f);
+    EXPECT_EQ(FP32{}.sign_bit(), 0);
+
+    EXPECT_EQ(FP64{}.as_uint(), 0);
+    EXPECT_EQ(FP64{}.as_float(), 0.0);
+    EXPECT_EQ(FP64{}.sign_bit(), 0);
+}
+
+TEST(floating_point_utils, fp_sign_bit)
```

Comment:
This test could also check positive/negative NaNs

Reply:
Added, with the details that creating negative NaNs reliably is only possible with FP's own negation operator.